internal/controller/services/servicemesh/servicemesh_support.go

```diff
@@ -54,11 +54,41 @@ func (r *ServiceMeshReconciler) configureServiceMesh(ctx context.Context, instan
 
 	case operatorv1.Unmanaged:
 		log.Info("ServiceMesh CR is not configured by the operator, we won't do anything")
+		// Update status ServiceMesh is unmanaged
+		conditions := instance.Status.Conditions
+		status.SetCondition(&conditions, status.CapabilityServiceMesh, status.MissingOperatorReason,
+			"ServiceMesh is Unmanaged in DSCI", metav1.ConditionFalse)
+		instance.Status.SetConditions(conditions)
+		if err := r.Client.Status().Update(ctx, instance); err != nil {
+			log.Error(err, "failed to update DSCI status condition for ServiceMesh")
+			return err
+		}
+
 	case operatorv1.Removed:
 		log.Info("existing ServiceMesh CR (owned by operator) will be removed")
 		if err := r.removeServiceMesh(ctx, instance); err != nil {
 			return err
 		}
+
+		// Remove condition if it was set when DSCI has Removed
+		conditions := instance.Status.Conditions
```

Comment:
In general, conditions should not be removed because that would mean the status is unknown, shouldn't this be `Status=False` ?

Reply:
so to make both Unmanaged and Removed the same condition?
at first, i set both to False on missingOperator, later changed it, because in case "create DSCI set original value to Removed." does not feel needed.

but i am fine with False when Removed.

Reply:
You can probably change the reason i.e. Status=False, Reason=Removed|Unmanaged